tutorials/11-probabilistic-pca/11_probabilistic-pca.jmd

```diff
@@ -155,52 +285,61 @@ df_rec[!, :cell] = 1:n_cells
+This is satisfying as we have just projected the original 9-dimensional space onto a 2-dimensional space - some info has been cut off in the projection process, but we haven't lost any important info, e.g. the key difference between the two groups.
+The is the desirable property of PCA: it picks up the principal axes along which most of the (original) data data variations cluster, and remove those less relevant.
+If we choose the reduced space dimension $k$ to be exactly $D$, we would recover exactly the same input matrix $x$, as the new (projected) space contains the same information as the input space as long as we keep all the dimensions.
 
-And finally, we plot the data in a lower dimensional space. The interesting insight here is that we can project the information from the input space into a two-dimensional representation, without lossing the essential information about the two groups of cells in the input data.
+Now we have represented the original high-dimensional data in two dimensions, without lossing the key information about the two groups of cells in the input data.
+Finally, the benefits of performing PCA is to analyse and visualise the dimension-reduced data in the projected, low-dimensional space.
 
+We can also check the average magnitudes of the gap between $\hat{x}$ and $x$:
 ```julia; echo=false
 let
-    diff = X' - expression_matrix
-    @assert mean(diff[:, 4]) < 0.5
-    @assert mean(diff[:, 5]) < 0.5
-    @assert mean(diff[:, 6]) < 0.5
+    diff_matrix = X' - expression_matrix
+    @assert abs(mean(diff_matrix[:, 4])) < 0.5
+    @assert abs(mean(diff_matrix[:, 5])) < 0.5
+    @assert abs(mean(diff_matrix[:, 6])) < 0.5
 end
 ```
 
+Finally, we save the projected results $z$ as a DataFrame, rename the columns and add an extra column for numbering cells.
+
 ```julia
 df_pca = DataFrame(z', :auto)
-rename!(df_pca, Symbol.(["z" * string(i) for i in collect(1:n_genes)]))
+rename!(df_pca, Symbol.(["z" * string(i) for i in collect(1:k)]))
 df_pca[!, :cell] = 1:n_cells
 
-@vlplot(:rect, "cell:o", "variable:o", color = :value)(DataFrames.stack(df_pca, 1:n_genes))
+@vlplot(:rect, "cell:o", "variable:o", color = :value)(DataFrames.stack(df_pca, 1:k))
+scatter(df_pca[:, :z1], df_pca[:, :z2]; xlabel="z1", ylabel="z2")
 
 df_pca[!, :type] = repeat([1, 2]; inner=n_cells ÷ 2)
 @vlplot(:point, x = :z1, y = :z2, color = "type:n")(df_pca)
 ```
 
-We can see the two groups are well separated in this 2-D space. Another way to put it, 2 dimensions is enough to display the main structure of the data.
+We see the two groups are well separated in this 2-D space. As an unsupervised learning method, performing PCA on this dataset gives membership for each cell instance.
+Another way to put it, 2 dimensions is enough to display the main structure of the data.
 
-## Number of components
+#### Further extension: automatic choice of the number of principal components with ARD
```

Comment:
```suggestion
#### Further extension: automatic choice of the number of principal components with Automatic Relevance Determination (ARD)
```
First time you introduce this term you should define instead of using acronyms.